src/utils/strings.ts

```diff
@@ -84,3 +86,24 @@ const collator = new Intl.Collator();
 export function compare(a: string, b: string): number {
     return collator.compare(a, b);
 }
+
+/**
+ * A cache for regular expression used in fuzzy matching
+ * @param str The string to inspect
+ * @return {RegExp} a regular expression to inspect `str`
```

Comment:
some words to explain that this is a performance benefit would be appreciated. Also fullstops :)